### DISCUSSION TITLE
SDL and OpenGL ES backend

### DISCUSSION THREAD

**Discussion by leonkasovan** [ikemen-engine/Ikemen-GO#973] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

We all love this engine. Most of us play in PC/laptop (windows, mac, linux). But there is also fans who play handheld game device (Anbernic, Miyoo, Retro Pocket etc). There is limitation when play in handheld device: 
1. doesn't support GLFW (which require windows manager, X server in linux)
2. doesn't support OpenGL (Desktop)

### Describe the solution you'd like

So that, the engine can be played in handheld device, we need:
1. support KMS/DRM via SDL2 backend
2. support renderer OpenGL ES backend

### Additional context

I would like to donate to this project ($10). Anybody who wants to contribute a new feature above, please feel free to contact me.
Best regards.

### Search terms

SDL SDL2 OpenGL OpenGLES GLES Mugen Ikemen Engine

**Comment by Lazin3ss** [comment]
Recently, GLFW and OpenGL were unhardcoded from the engine by Sam, so adding SDL as backend is more feasible now. Hopefully, in the future we'll be able to do it.

Also, we devs don't accept donations in general, but we appreciate the gesture.

**Comment by lovelydumpling** [comment]
As a recent owner of the miyoo mini, rg351p, and rg353v, I would love being able to have IKEMEN on these devices. Android as well for the Odin Lite

**Reply by leonkasovan** [reply]
Mine : RG353P 😄
Such a pleasure to watch ikemen-go running on that device

**Comment by Ultrahead** [comment]
Is there any progress on this? I have a TSP where neither X11 nor wayland is present, so a solution based purely on SDL2 / GLES is welcome.

**Reply by Ultrahead** [reply]
@Lazin3ss ?

**Reply by leonkasovan** [reply, marked as answer]
> Is there any progress on this? I have a TSP where neither X11 nor wayland is present, so a solution based purely on SDL2 / GLES is welcome.

Your TSP Firmware support PortMaster?
Now, you can install Ikemen via PortMaster App.

Or download Ikemen package manually from:
https://portmaster.games/games.html

**Reply by Ultrahead** [reply]
Done. Thanks!
